lib/ClientGeneratorFromXml.php

```diff
@@ -292,7 +306,9 @@ protected function loadTypesRecursive($type, $strict = true)
 		$classNodes = $xpath->query("/xml/classes/class[@base = '$type']");
 		foreach($classNodes as $classNode)
 		{
-			$this->loadTypesRecursive($classNode->getAttribute("name"));
+			if ($this->shouldExtendType($type)) {
```

Comment:
You could place this check before you query the xPath and before the foreach loop, although I'm not concerned by performance in this tool I think it would be the right thing to do.

Reply:
Makes sense, fixed